### PR TITLE
chore: update CAS section with warning

### PR DESCRIPTION
**Pull request by pratik151192**

### CODE REVIEW

File: docs/cache/develop/api-reference/index.mdx [hunk 1]
```diff
@@ -261,6 +261,11 @@ See [response objects](./response-objects.md) for specific information.
 
 ### SetIfAbsent
 
+:::warning
+Do NOT use compare-and-set (CAS) APIs such as `SetIfAbsent` with scalar or non-CAS APIs such as `Set`. Doing so will result
```
Comment (krispraws) [inline docs/cache/develop/api-reference/index.mdx]: Can we call out `Delete` also as a non-CAS API here?

**Review by krispraws — COMMENTED** [review]
We are missing the warning for SetIfEqual and SetIfNotEqual.

**Review by krispraws — DISMISSED** [review]
Nitpicky - I have heard of Redis check-and-set, and compare-and-swap. Is compare-and-set widely used?

**Comment by pratik151192** [comment]
@krispraws eww you're right; CAS traditionally in my mind comes from CPU provided CAS instruction which is compare-and-swap. updated

**Comment by krispraws** [comment]
> @krispraws eww you're right; CAS traditionally in my mind comes from CPU provided CAS instruction which is compare-and-swap. updated

Aren't we closer to the Redis terminology though? `swap` to me implies atomicity, which we cannot guarantee.

**Comment by pratik151192** [comment]
@krispraws  TIL Redis `check-and-set`.. updated!

**Review by krispraws — APPROVED** [review]

**Pull request merged**